### PR TITLE
fix_: add ticker to check peers and update connection status

### LINKED ISSUE #5679: Rapid log file growth and repeated Waku errors when DNS encryption is not set
### Issue Description

When running the Status app without setting encrypted DNS (e.g., https://9.9.9.9/dns-query), the geth.log file grows rapidly within seconds of opening the app. The log is filled with repetitive error messages, primarily related to Waku and peer selection.

### Observed Behavior

1. Without encrypted DNS, opening the Status app causes geth.log to generate a large volume of repeated error logs within seconds.
2. The log file size increases rapidly, potentially overwriting initial logs (e.g., 'status-go version details').
3. The majority of error messages are related to peer selection and Waku.

### Log Analysis

The log shows multiple recurring errors:
- "peer selection returned err"
- "no suitable peers found"
- "dns discovery error"
- "could not obtain dns discovery peers"

Example log snippet:
```
ERROR[08-08|08:12:56.847][github.com/status-im/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/filter/client.go:350] peer selection returned err
error="no suitable peers found"
```

### Potential Causes

1. GFW (Great Firewall) interference: The issue may be related to network restrictions in certain regions.
2. Waku-related code issues in the Status app.
3. DNS resolution problems: Default DNS may fail to resolve Ethereum network node addresses correctly.
4. Network connectivity issues: Certain network configurations may be blocking the Status app from connecting to the Ethereum network.

### Workaround

Setting encrypted DNS (e.g., https://9.9.9.9/dns-query) appears to resolve or mitigate the issue. This suggests that the problem might be related to DNS resolution or network restrictions.

### Additional Context

- The issue seems to create a loop-like behavior in Waku-related processes.
- The problem may be more pronounced in regions with strict internet controls.

### Questions for Investigation

1. Is there a potential infinite loop in the Waku-related code when peer discovery fails?
2. How does the encrypted DNS setting affect the peer discovery process?
3. Are there any network-level checks that can be implemented to gracefully handle situations where peer discovery consistently fails?

### Environment

- Status app version: on branch `develop`, commit id: `3b4cbfa4bca1b76cf3e9675e2e6b99d14c09b9e1`
- ios simulator: iphone 15 pro
- [surge](https://nssurge.com/) v5.7.x

### Steps to Reproduce

1. Ensure encrypted DNS is not set in surge.
2. Open the Status app.
3. Monitor the geth.log file for rapid growth and repeated error messages.

relate screenshots and comments:

> This might be a serious issue. I just opened the Status app for a few seconds, and the log file size grows rapidly. The content seems to repeatedly relate to Waku, as if there is a dead loop. I can't even see the initial log like status-go version details because it might be getting overwritten.

image

> after changing log level to INFO , geth.log get fullfilled with this content, so now I'm sure there's something relate to waku

image

update encrypted DNS

### PR DESCRIPTION
**Pull request by chaitanyaprem**
Looking at debug logs shared , i think the issue seems to be that node state is set to `online` from the application once it detects there is a network connection.  This starts all underlying process of subscribing for filters which keeps failing as no peers are available. 
After that since no peers are detected, node state is not changed to offline as currently that relies on peer connection notifications. 
there are 2 things we can do: one of them is fixed in this PR.

- [x] bring back the ticker in status-go that checks for peerCount and decides if node is online or not. This should prevent from immediate loop of retries for Filter subscribe.

we can add some backoff to filter subscribe in case of subsequent failures especially for specific errors such as "no suitable peers found" . This would prevent such situation from occuring. Will be addressed in separate PR.

Closes #5679

### CODE REVIEW

File: wakuv2/waku.go [hunk 1]
```diff
@@ -1087,51 +1087,18 @@ func (w *Waku) Start() error {
 
 	go func() {
 		defer w.wg.Done()
-
+		ticker := time.NewTicker(5 * time.Second)
```
Comment (kaichaosun) [inline wakuv2/waku.go:1094]: Can we make it a constant? Also 5 seconds seems pretty frequent.

**Review by kaichaosun — APPROVED** [review]

**Review by qfrank — APPROVED** [review]
thank you for the fix, it works 👍

File: wakuv2/waku.go (hunk 1 above)
Comment (chaitanyaprem) [reply wakuv2/waku.go:1094]: Hmm, previous ticker also used to 5sec, so i just went with that. any suggestions or thoughts as to what can be optimal value?

Comment (kaichaosun) [reply wakuv2/waku.go:1094]: I usually starts with a slower value and make it faster when requires by people. It's probably fine here as you are using the existing ticker.

**Comment by chaitanyaprem** [comment]
I think we can merge this as this is not affecting anything else and i have dogfooded relay as well using desktop.

@richard-ramos do need your approval though.

**Review by richard-ramos — APPROVED** [review]

**Comment by qfrank** [comment]
> I think we can merge this as this is not affecting anything else and i have dogfooded relay as well using desktop.
> 
> @richard-ramos do need your approval though.

For mobile team, If there's no need for manual QA and only status-go changes, we'll create a mobile PR and tag with `skip-manual-qa` and `status-go` and add a comment cc @status-im/mobile-qa to ask for a checking for the e2e test result. If the result looks good, we can merge it then, I'm curious what's @churik 's thoughts on this 🙂 

ps: I'm not saying I don't trust you @chaitanyaprem , just curious 😅

**Comment by chaitanyaprem** [comment]
> > I think we can merge this as this is not affecting anything else and i have dogfooded relay as well using desktop.
> > @richard-ramos do need your approval though.
> 
> For mobile team, If there's no need for manual QA and only status-go changes, we'll create a mobile PR and tag with `skip-manual-qa` and `status-go` and add a comment cc @status-im/mobile-qa to ask for a checking for the e2e test result. If the result looks good, we can merge it then, I'm curious what's @churik 's thoughts on this 🙂
> 
> ps: I'm not saying I don't trust you @chaitanyaprem , just curious 😅

created a mobile PR https://github.com/status-im/status-mobile/pull/21037 so that e2e can be run. Once e2e is cleared, will merge this.

**Comment by pavloburykh** [comment]
Hey @qfrank @chaitanyaprem! Thanx for the PR. I have just triggered e2e run manually here https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/4308/ because in this case (PR containing go) we need to run a full testrun. 

When testrun is triggered automatically we are triggering just a smoke run with limited amount of cases which is not enough for PRs containing go changes. 

Please, ping @yevh-berdnyk  e2e once results are posted here so she can analyse them before merge. Thank you!

**Pull request merged**